synapse/federation/sender/__init__.py

```diff
@@ -275,15 +275,27 @@ async def _process_event_queue_loop(self) -> None:
                 if not events and next_token >= self._last_poked_id:
                     break
 
-                async def handle_event(event: EventBase) -> None:
+                async def get_destinations_for_event(
+                    event: EventBase,
+                ) -> Optional[Collection[str]]:
```

Comment:
you could have it return an empty tuple when there are no destinations. Feels more elegant/consistent to me. YMMV.

Reply:
I agree, but it would make checking for "do not send this event to any destination" much more cleaner by check of `is not None`, which is more explicit (imo) than `len(ret) > 0`, same with the current `None` returns, because those'd have to reply with a `Collection`, for which i find `[]`, `set()`, and `tuple()` all not very appealing to semantically read.

It *would* make it cleaner to return a `Collection` in this return type, but i think it'd suffer the readability for it overall, for me personally.

Reply:
> I agree, but it would make checking for "do not send this event to any destination" much more cleaner by check of `is not None`, which is more explicit (imo) than `len(ret) > 0`, same with the current `None` returns, because those'd have to reply with a `Collection`, for which i find `[]`, `set()`, and `tuple()` all not very appealing to semantically read.

I have no idea how to parse this sentence.

Reply:
by the way, remember that empty lists and tuples are falsey, so rather than `if len(ret) > 0:`, you can just write `if ret:`.

Reply:
> I have no idea how to parse this sentence.

TL;DR: I think it'll look less pythonic because of all the `return []`s and/or the like

Reply:
I am unconvinced that `return ()` is "less pythonic"  than `return None`, and in any case personally I find it at least as readable.

Reply:
...D'oh, i thought `()` != "empty tuple", mainly cuz Rust uses this as their "empty" type, i got confused for a second.

In that case, i think using `return ()` is perfectly fine, i messed it up, i'll change this to that :+1: